File: backend/src/main/java/com/example/backend/entity/Message.java
package com.example.backend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("message")
public class Message {

    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("conversation_id")
    private Long conversationId;

    // ai or user
    @TableField("role")
    private String role;

    @TableField("content")
    private String content;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: video-mind-web/src/types/message.ts
/**
 * 消息实体
 */
export interface Message{
    id: string
    role: 'user' | 'ai'
    content?: string
    rawContent?: string
    isPlaceholder?: boolean
    placeholderType?: 'summary' | 'chat'
    isStreaming?: boolean
}